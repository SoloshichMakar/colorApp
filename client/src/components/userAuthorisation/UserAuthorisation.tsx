import React from "react";
import { Redirect, NavLink } from "react-router-dom";
import TasksContainer from "../taskContainer/TasksContainer";

interface IUserAuthorisationProps {
  email: string;
  password: string;
  isAuthenticated: boolean;
  message: string;
  textEmailChange: Function;
  textPasswordChange: Function;
  userValidate: Function;
}

const UserAuthorisation: React.FC<IUserAuthorisationProps> = ({
  email,
  password,
  isAuthenticated,
  message,
  textEmailChange,
  textPasswordChange,
  userValidate,
}) => {
  function currentTextEmailChange(e: React.ChangeEvent<HTMLInputElement>) {
    const getTextAreaValue = e.target.value;
    textEmailChange(getTextAreaValue);
  }

  function currentTextPasswordChange(e: React.ChangeEvent<HTMLInputElement>) {
    const getTextAreaValue = e.target.value;
    textPasswordChange(getTextAreaValue);
  }

  function UserAuthorisationCheck() {
    userValidate(email, password);
  }

  if (isAuthenticated) {
    return <Redirect to="/" />;
  }
  return (
    <div className="login_main">
      <div>Log in Page</div>
      <div className="login_main__data">
        <div className="login_main__message_box">
          <div className="login_main__message_label">{message}</div>
        </div>
        <div className="login_main__email_box">
          <div className="login_main__email_label">Email</div>
          <input
            className="login_main__email_input"
            type="text"
            onChange={currentTextEmailChange}
          />
        </div>
        <div className="login_main__password_box">
          <div className="login_main__password_label">Password</div>
          <input className="login_main__password_input" type="password" onChange={currentTextPasswordChange} />
        </div>
      </div>
      <input className="login_main__login_button" type="button" value="Log in" onClick={UserAuthorisationCheck} />
      <NavLink className="login_main__registration_link" to="/registration">Sign in</NavLink>
    </div>
  );
};

export default UserAuthorisation;
