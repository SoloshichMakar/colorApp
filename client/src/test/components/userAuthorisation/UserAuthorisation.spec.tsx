import React from 'react';
import renderer from 'react-test-renderer';

import UserAuthorisation from '../../../components/userAuthorisation/UserAuthorisation';
import {BrowserRouter} from "react-router-dom";


const props = {
    isAuthenticated: false,
    message: 'Test Messsage',
    email: 'Test@mail.com',
    password: 'Test Password',
    textEmailChange: Function,
    textPasswordChange: Function,
    userValidate: Function,
};


it('renders correctly', () => {
    const tree = renderer
        .create(
            <BrowserRouter>
                <UserAuthorisation
                    isAuthenticated={props.isAuthenticated}
                    message={props.message}
                    email={props.email}
                    password={props.password}
                    textEmailChange={props.textEmailChange}
                    textPasswordChange = {props.textPasswordChange}
                    userValidate={props.userValidate}/>
            </BrowserRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});