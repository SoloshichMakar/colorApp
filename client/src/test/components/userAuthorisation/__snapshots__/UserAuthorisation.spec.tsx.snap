// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="login_main"
>
  <div>
    Log in Page
  </div>
  <div
    className="login_main__data"
  >
    <div
      className="login_main__message_box"
    >
      <div
        className="login_main__message_label"
      >
        Test Messsage
      </div>
    </div>
    <div
      className="login_main__email_box"
    >
      <div
        className="login_main__email_label"
      >
        Email
      </div>
      <input
        className="login_main__email_input"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="login_main__password_box"
    >
      <div
        className="login_main__password_label"
      >
        Password
      </div>
      <input
        className="login_main__password_input"
        onChange={[Function]}
        type="password"
      />
    </div>
  </div>
  <input
    className="login_main__login_button"
    onClick={[Function]}
    type="button"
    value="Log in"
  />
  <a
    aria-current={null}
    className="login_main__registration_link"
    href="/registration"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
